### DISCUSSION TITLE
PIVOT should have syntax to GROUP BY no columns

### DISCUSSION THREAD

**Discussion by DanCory** [duckdb/duckdb#12673] [Ideas]
### What happens?

You want to PIVOT your data but only have one row of output with data in columns. There's no syntax to indicate that any extra fields should be ignored. The default behavior of PIVOT is to group by all other fields (GROUP BY ALL). The PIVOT case needs something like GROUP BY NONE or just an empty GROUP BY.

### To Reproduce

```sql
CREATE TABLE Cities (
    Country VARCHAR, Name VARCHAR, Year INTEGER, Population INTEGER
);
INSERT INTO Cities VALUES ('NL', 'Amsterdam', 2000, 1005);
INSERT INTO Cities VALUES ('NL', 'Amsterdam', 2010, 1065);
INSERT INTO Cities VALUES ('NL', 'Amsterdam', 2020, 1158);
INSERT INTO Cities VALUES ('US', 'Seattle', 2000, 564);
INSERT INTO Cities VALUES ('US', 'Seattle', 2010, 608);
INSERT INTO Cities VALUES ('US', 'Seattle', 2020, 738);
INSERT INTO Cities VALUES ('US', 'New York City', 2000, 8015);
INSERT INTO Cities VALUES ('US', 'New York City', 2010, 8175);
INSERT INTO Cities VALUES ('US', 'New York City', 2020, 8772);

PIVOT Cities
ON Year
USING sum(Population);
-- This groups by Country and Name

-- I would like to just get one row with a column for each year.
-- The shortest workaround I know of is
with step1 as (select sum(Population) as Population from Cities group by Year),
pivot step1 on Year using sum(Population);
```

### OS:

Windows amd64

### DuckDB Version:

1.0.0 main

### DuckDB Client:

CLI

### Full Name:

Dan Cory

### Affiliation:

self

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by chanon** [comment]
Here is a related feature request I just made that also covers this:
https://github.com/duckdb/duckdb/discussions/12860

**Comment by rpbouman** [comment]
Not saying this shouldn't be a feature, but there's a workaround that doesn't seem too bad to me:

```
PIVOT (FROM Cities CROSS JOIN (VALUES(NULL)) AS T(one_group)) 
ON Year 
USING SUM(population) 
GROUP BY one_group;
```

or maybe this

```
PIVOT (SELECT *, NULL AS one_group FROM Cities)
ON Year 
USING SUM(population) 
GROUP BY one_group;
```

```
┌───────────┬────────┬────────┬────────┐
│ one_group │  2000  │  2010  │  2020  │
│   int32   │ int128 │ int128 │ int128 │
├───────────┼────────┼────────┼────────┤
│    <NULL> │   9584 │   9848 │  10668 │
└───────────┴────────┴────────┴────────┘
```
